lib/client.js

```diff
@@ -384,13 +384,6 @@ module.exports = function (dependencies) {
         this.destroySession(session);
       });
 
-      this.session.on('socketError', error => {
```

Comment:
I removed this as it's no longer available in the older version of the node tested ([version 14 LTS](https://nodejs.org/docs/latest-v14.x/api/http2.html)). The last time I see it is in [node version 8.7.0](https://nodejs.org/download/release/v8.7.0/docs/api/http2.html).

I'm sure whatever it use to catch is still caught by the rest of the error handling in this promise